src/test/java/util/xcmailr/util/SendRequest.java

```diff
@@ -104,32 +111,39 @@ public static void deleteTempEmail(String tempEmail)
             e.printStackTrace();
             throw new RuntimeException(e);
         }
+    }
 
+    public static boolean emailExists(String email)
+    {
+        return getTempEmail(email) != null;
     }
 
-    private static String getIdOfTempEmail(String tempEmail)
+    public static boolean emailExpired(String email)
     {
+        return getTempEmail(email).get("expired").getAsBoolean();
+    }
+
+    public static void deleteTempEmail(String tempEmail)
+    {
+        String id = getTempEmail(tempEmail).get("id").getAsString();
         HttpResponse response = null;
         StringBuilder sb = new StringBuilder();
         BufferedReader br = null;
         try
         {
-            HttpGet getRequest = new HttpGet("https://xcmailr.xceptance.de/mail/getmails");
+            HttpPost postRequest = new HttpPost("https://xcmailr.xceptance.de/mail/delete/" + id);
+            response = httpClient.execute(postRequest);
 
-            response = httpClient.execute(getRequest);
             br = new BufferedReader(new InputStreamReader(response.getEntity().getContent(), "UTF-8"));
-
-            String output;
-            while ((output = br.readLine()) != null)
+            String output = br.readLine();
```

Comment:
While this does indeed solve the error we had previously, `br` can still be `null` if the `httpClient.execute` method failed.
Hence, you should add a null check to the `br.close` in line 150.

However, why do we only close the `BufferedReader` when an exception is thrown?